ttnn/cpp/ttnn/tensor/layout/tensor_layout.cpp

```diff
@@ -122,33 +159,104 @@ size_t TensorLayout::compute_page_size_bytes(const Size& page_size) const {
     return page_config_.get_page_size_bytes(page_size, dtype_);
 }
 
+Size TensorLayout::compute_physical_shard_shape(const Size& logical_shard_shape) const {
```

Comment:
To discuss later. TensorLayout will become a permutation of everything as contains logic for the handling of sharded and non-sharded cases. Want to discuss how we can abstract this away. Maybe similarly to what we did with page config.